### PR TITLE
feat: Add "js" propset type support

### PR DESCRIPTION
**Pull request by tchock**

### CODE REVIEW

File: packages/cli/src/commands/bundle.ts [hunk 1]
```diff
@@ -61,23 +63,26 @@ export const bundleAction = ({
       componentExtractStep.success(getComponentCountText(componentCount, fileCount));
 
       const decoratorFiles = await glob(decoratorPath, { cwd: executionPath });
+      const storyFiles = await glob(storyPath, { cwd: executionPath });
```
Comment (JohannesMerz) [inline packages/cli/src/commands/bundle.ts]: need to add this to our config docs

File: packages/cli/src/commands/bundle.ts [hunk 2]
```diff
@@ -61,23 +63,26 @@ export const bundleAction = ({
...
       const decoratorFileArray =
         decoratorFiles.length > 0 ? [pathUtils.resolve(executionPath, decoratorFiles[0])] : [];
+      const storyFileArray = storyFiles.map(sf => pathUtils.resolve(executionPath, sf));
 
-      const entrypoints = Object.entries(entrypointsWithMetadata).reduce(
-        (prev, [key, ep]) => ({
+      const entrypoints = Object.entries(entrypointsWithMetadata).reduce((prev, [key, ep]) => {
+        const abc = storyFileArray.filter(sf => sf.includes(key));
```
Comment (JohannesMerz) [inline packages/cli/src/commands/bundle.ts]: abc? can we rename this to sth that means sth?

File: packages/cli/src/utils/exposeLoader.ts [hunk 3]
```diff
@@ -44,9 +44,12 @@ export const pitch = function(remainingRequest) {
...
   // eslint-disable-next-line no-underscore-dangle
-  this._module.userRequest = `${this._module.userRequest}-exposed`;
-  return (
-    `${accesorString(this.query.substr(1))} = ` +
-    `require(${JSON.stringify(`-!${newRequestPath}`)});`
-  );
+  const originalUserRequest = this._module.userRequest;
+  // eslint-disable-next-line no-underscore-dangle
+  this._module.userRequest = `${originalUserRequest}-exposed`;
+  const symbol =
```
Comment (JohannesMerz) [inline packages/cli/src/utils/exposeLoader.ts]: can you ELI5 what we are doing here?

File: packages/collector-base/src/propsRegistry.ts [hunk 4]
```diff
@@ -2,23 +2,33 @@ import { creatFilePropsKey, createFilePropList } from './createFilePropList';
 
 export const propsRegistry = new Map();
 
-export type AddPropsFn = (
-  filePath: string,
-  exportName: string,
-  propSet: Record<string, any>,
-  name?: string
-) => void;
+export type PropSetType = 'json' | 'js';
```
Comment (JohannesMerz) [inline packages/collector-base/src/propsRegistry.ts]: whats the diff to an enum?

File: packages/cli/src/commands/bundle.ts (hunk 2 above)
Comment (tchock) [reply packages/cli/src/commands/bundle.ts]: Ah. damn. That logic is even wrong :D

File: packages/collector-base/src/propsRegistry.ts (hunk 4 above)
Comment (tchock) [reply packages/collector-base/src/propsRegistry.ts]: `enum` is like `PropSetType.json`.
This one is just a plain string but it is validated with TypeScript.

File: packages/cli/src/utils/exposeLoader.ts (hunk 3 above)
Comment (tchock) [reply packages/cli/src/utils/exposeLoader.ts]: This is mostly copy paste from `expose-loader` library.
`symbol` comes from the `options` in webpack.
it can either be a function that receives the file path or it's a string.

This is to define the name for the variable that is exposed in `window`.

File: packages/cli/src/commands/bundle.ts (hunk 2 above)
Comment (tchock) [reply packages/cli/src/commands/bundle.ts]: done
